service/src/main/java/bio/terra/workspace/service/resource/ResourceValidationUtils.java

```diff
@@ -421,17 +437,30 @@ public static void validateAzureVmSize(String vmSize) {
     }
   }
 
-  public static void validateRegion(String region) {
+  public static void validateAzureRegion(String region) {
     if (!Region.values().stream()
         .map(Region::toString)
         .collect(Collectors.toList())
         .contains(region)) {
       logger.warn("Invalid Azure region {}", region);
-      throw new InvalidReferenceException(
+      throw new InvalidControlledResourceException(
           "Invalid Azure Region specified. See the class `com.azure.core.management.Region`");
     }
   }
 
+  public void validateGcpRegion(UUID workspaceId, String region) {
+    region = GcpUtils.parseRegion(region);
+
+    // Get the list of valid locations for this workspace from TPS. If there are no regional
+    // constraints applied to the workspace, TPS should return all available regions.
+    List<String> validLocations = tpsApiDispatch.listValidRegions(workspaceId, "gcp");
+
+    if (validLocations.stream().noneMatch(region::equalsIgnoreCase)) {
```

Comment:
Should we normalize regions to lowercase when they are assigned to the resource?
Otherwise, we have to always be remembering to ignore case and will return improperly cased region names, right?

We could have this method return the canonical region name rather than whatever the input was. wdyt?

Reply:
I've been thinking of where to do this and this probably isn't the best spot for it. When we call this, we've already put together the creation params into an object which is now read-only.
In the end, GCP doesn't seem to care what the case is on the resource. There's also some regions that are canonically upper-case in GCP (IE: multi-region = US or EU).
I think we could normalize the region against the generic list of all available regions earlier on, like when we put the common params together, but that feels like it belongs in a separate work item.